src/main/flight/failsafe.c

```diff
@@ -146,19 +139,6 @@ static bool failsafeShouldHaveCausedLandingByNow(void)
     return (millis() > failsafeState.landingShouldBeFinishedAt);
 }
 
-static void failsafeActivate(void)
```

Comment:
Removing this function repeating this code 5 times is not the right approach.
You could add a parameter to use either FAILSAFE_MODE or GPS_RESCUE_MODE.
Remove the line 
```
failsafeState.landingShouldBeFinishedAt = millis() + failsafeConfig()->failsafe_off_delay * MILLIS_PER_TENTH_SECOND;
```
as this is now used in one place only.

Put the line 
```
failsafeState.receivingRxDataPeriodPreset = failsafeState.rxDataRecoveryPeriod;
```
in this function.

Reply:
On some places `failsafeState.*` are altered. `failsafeActivate` do only part of this job. Maybe we can think of different encapsulation.

Using `failsafeActivate` and then manually alter `failsafeState.*` on top of that is not better solution from my point of view.